backend/src/main/java/com/ambacam/model/Operateur.java

```diff
@@ -0,0 +1,140 @@
+package com.ambacam.model;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+@Entity
+public class Operateur extends AuditingCommonEntity implements Serializable {
+	private static final long serialVersionUID = 7327701821300391989L;
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@NotNull
+	@NotEmpty
+	@Column(columnDefinition = "varchar", nullable = false)
+	private String nom;
+
+	@Column(columnDefinition = "varchar", nullable = true)
+	private String prenom;
+
+	@NotNull
+	@NotEmpty
+	@Column(columnDefinition = "varchar", nullable = false)
+	private String sexe;
+
+	@ManyToOne
+	@JoinColumn(name = "nationalite_id")
+	private Pays nationalite;
+
+	@NotNull
+	@NotEmpty
+	@Column(columnDefinition = "varchar", nullable = false)
+	private String username;
+
+	@NotNull
+	@NotEmpty
+	@Column(columnDefinition = "varchar", nullable = false)
+	private String password;
+
+	@ManyToOne
+	@JoinColumn(name = "cree_par_id")
+	private Operateur creePar;
+
+	@ManyToMany
+	@JoinTable(name = "operateur_role", joinColumns = @JoinColumn(name = "operateur_id", nullable = false, updatable = false), inverseJoinColumns = @JoinColumn(name = "role_id", nullable = false, updatable = false))

```

Comment:
I think this should be updatable since this parameter is not required in the creation

Reply:
I understand the point and it is also logic, but this represents the pair (operateur_id, role_id) in the additional table created automatically by JPA. Therefore, it is not nullable not updatable since each new relationship Operateur-Role (including existing connexions) will be a new entry.